etl/transform/datadiscovery_cards.py

```diff
@@ -0,0 +1,316 @@
+    """
+    Get/Generate URI from BrAPI object or generate one
+    """
+    pui_field = entity + 'PUI'
+    #TODO: this is going to be problematic since in GnpIS studies are using germplasmDbIb(num) and not germplasmDbIb(DOI)
+    #TODO (cont): consider using a fully generated dbId, using urn, no matter what.
+    #TODO (cont): should be ok, check with Célia, Cyril, Maud, Nico ?
+    data_uri = data.get(pui_field)
+
+    if data_uri and rfc3987.match(data_uri, rule='URI'):
+        # The original PUI is a valid URI
+        return data_uri
+
+    source_id = urllib.parse.quote(source['schema:identifier'])
+    data_id = get_identifier(entity, data)
+    if not data_uri:
+        # Generate URI from source id, entity name and data id
+        encoded_entity = urllib.parse.quote(entity)
+        encoded_id = urllib.parse.quote(data_id)
+        data_uri = f"urn:{source_id}/{encoded_entity}/{encoded_id}"
+    else:
+        # Generate URI by prepending the original URI with the source identifier
+        encoded_uri = urllib.parse.quote(data_uri)
+        data_uri = f"urn:{source_id}/{encoded_uri}"
+    if not rfc3987.match(data_uri, rule='URI'):
+        raise Exception(f'Could not get or create a correct URI for "{entity}" object id "{data_id}"'
+                        f' (malformed URI: "{data_uri}")')
+    return data_uri
+
+
+def load_input_json(source, doc_types, source_json_dir, config):
+    data_dict = {}
+    if source_json_dir:
+        #all_files = list_entity_files(source_json_dir)
+        #filtered_files = list(filter(lambda x: x[0] in source_entities, all_files))
+        for document_type in doc_types:
+            if document_type["document-type"] == "observationUnit":
+                pass
+            input_json_filepath = source_json_dir + "/" + document_type["document-type"] + ".json"
+            data_dict[document_type["document-type"]] = {}
+            try:
+                with open(input_json_filepath, 'r') as json_file:
+                    json_list = list(json_file)
+                    for json_line in json_list:
+                        data = json.loads(json_line)
+                        uri = get_generated_uri(source, document_type["document-type"], data)
+                        data_dict[document_type["document-type"]][uri] = data
+    #                    links = get_entity_links(data, 'DbId')
+    #                    entity_names = set(map(first, links))
+            except FileNotFoundError as e:
+                print("No "+document_type["document-type"]+" in "+source['schema:identifier'])
+
+    return data_dict
+
+
+def set_dbid_to_uri(data_dict,source):
+
+    dbid_to_uri = [("germplasmDbId","germplasm"), ("locationDbId","location"), ("studyDbId","study"), ("trialDbId","trial")]
+    dbids_to_uri = [("germplasmDbIds","germplasm"), ("locationDbIds","location"), ("studyDbIds","study"), ("trialDbIds","trial")]
+
+    for dbids in dbids_to_uri :
```

Comment:
suggested refactor: rename that variable to make it more explicit, something like : dbids_by_type or something like that